dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsConnector.java

```diff
@@ -37,12 +38,31 @@ public OracleStatsConnector() {
   @Override
   public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments) throws Exception {
     StatsTaskListGenerator taskListGenerator = new StatsTaskListGenerator();
-    out.addAll(taskListGenerator.createTasks(arguments));
+    int queryLogDays = getQueryLogDays(arguments);
+    out.addAll(taskListGenerator.createTasks(arguments, queryLogDays));
   }
 
   @Nonnull
   @Override
   public String summary(String fileName) {
     return String.format("Oracle statistics saved to %s", fileName);
   }
+
+  static int getQueryLogDays(ConnectorArguments arguments) {
+    Integer asObject = arguments.getQueryLogDays();
+    int shortTimeInDays = 7;
+    int longTimeInDays = 30;
+    if (asObject == null || asObject == longTimeInDays) {
+      return longTimeInDays;
+    } else if (asObject == shortTimeInDays) {
+      return shortTimeInDays;
+    } else {
+      throw invalidDuration(asObject.intValue());
```

Comment:
I would recommend directly throwing the exception instead of delegating it to a method. The additional indirection seems unnecessary.